streams/src/main/java/org/apache/kafka/streams/errors/TaskMigratedException.java

```diff
@@ -21,8 +21,12 @@
 import org.apache.kafka.streams.processor.TaskId;
 
 /**
- * Indicates that a task got migrated to another thread.
- * Thus, the task raising this exception can be cleaned up and closed as "zombie".
+ * Indicates that one or more tasks got migrated to another thread.
+ *
+ * 1) if the task field is specified, then that single task should be cleaned up and closed as "zombie" while the
```

Comment:
This is the proposed exception semantics.

Reply:
Can you elaborate on when which version of `TaskMigratedException` should be used? ie, when is it possible for just one task to be a zombie while all the others can/should continue as normal

Reply:
It is summarized in https://confluentinc.atlassian.net/browse/KSTREAMS-3302. Note it is just one proposal and we can debate whether we agree or disagree about that :)

Reply:
I see, that makes sense. I have some thoughts on the handling in that case but I'll save it for the sync

Reply:
We can do this in a follow-up PR, but just to be clear we should always treat a `TaskMigratedException` as applying to _all_ tasks, and close them all as zombies. There's no possible way for one task to be a zombie while the others are able to continue processing